### ISSUE TITLE
Why isn't Remote DNS working?

### ISSUE THREAD

**Issue by ndiruhniu** [Mahdi-zarei/nekoray#205]
I have these parameters:
![pic1](https://github.com/user-attachments/assets/70686166-1f0b-40b5-a916-8ddb325d430b)

And only the wireguard protocol uses them. For example, a couple of lines from the log when connecting via wireguard:

INFO[0404] [3775256041 0ms] inbound/mixed[mixed-in]: inbound connection from 127.0.0.1:55022
INFO[0404] [3775256041 0ms] inbound/mixed[mixed-in]: inbound connection to github.com:443
INFO[0404] [3775256041 1ms] outbound/wireguard[proxy]: outbound connection to 116.202.176.26:853
INFO[0404] [3775256041 213ms] dns: exchanged github.com A github.com. 51 IN A 140.82.121.4
INFO[0404] [3775256041 214ms] dns: lookup succeed for github.com: 140.82.121.4
INFO[0404] [3775256041 214ms] outbound/wireguard[proxy]: outbound connection to 140.82.121.4:443
INFO[0405] [4125654795 0ms] inbound/mixed[mixed-in]: inbound connection from 127.0.0.1:55024
INFO[0405] [4125654795 0ms] inbound/mixed[mixed-in]: inbound connection to api.github.com:443
INFO[0405] [4125654795 1ms] outbound/wireguard[proxy]: outbound connection to 116.202.176.26:853
INFO[0406] [4125654795 329ms] dns: exchanged api.github.com A api.github.com. 24 IN A 140.82.121.6
INFO[0406] [4125654795 329ms] dns: lookup succeed for api.github.com: 140.82.121.6
INFO[0406] [4125654795 329ms] outbound/wireguard[proxy]: outbound connection to 140.82.121.6:443

And now via shadowsocks:

INFO[0003] [1880060277 0ms] inbound/mixed[mixed-in]: inbound connection from 127.0.0.1:41704
INFO[0003] [1880060277 0ms] inbound/mixed[mixed-in]: inbound connection to github.com:443
INFO[0003] [1880060277 1ms] outbound/shadowsocks[proxy]: outbound connection to github.com:443
INFO[0004] [3964320041 0ms] inbound/mixed[mixed-in]: inbound connection from 127.0.0.1:41712
INFO[0004] [3964320041 0ms] inbound/mixed[mixed-in]: inbound connection to api.github.com:443
INFO[0004] [3964320041 1ms] outbound/shadowsocks[proxy]: outbound connection to api.github.com:443

Am I doing something wrong? If so, what?

**Comment by ndiruhniu** [comment]
But why is nothing resolving at all? It also causes a DNS leak.

**Comment by ndiruhniu** [comment]
Let me rephrase the question: how can I use my DNS on other protocols?

**Comment by Mahdi-zarei** [comment]
> Let me rephrase the question: how can I use my DNS on other protocols in proxy mode (since in Tun Mode the DNS works fine)?

Set inbound domain strategy.

**Comment by ndiruhniu** [comment]
Yes, that's how it works, but I still don't understand why. That is, why in Tun Mode inbound domain strategy doesn't need to be touched and DNS requests will go through, but not in Proxy?
